monai/utils/misc.py

```diff
@@ -916,3 +916,16 @@ def unsqueeze_right(arr: NT, ndim: int) -> NT:
 def unsqueeze_left(arr: NT, ndim: int) -> NT:
     """Prepend 1-sized dimensions to `arr` to create a result with `ndim` dimensions."""
     return arr[(None,) * (ndim - arr.ndim)]
+
+
+def flatten_dict(metrics: dict[str, Any]) -> dict[str, Any]:
```

Comment:
I think there is a potential risk when the input dict has the same key in different dict level. For example:
```
input_dict = {"a":1, "b":2, "c": {"a":3, "d":4}}
print(flatten_dict(input_dict))
```

The output will be:
```
{'a': 3, 'b': 2, 'd': 4}
```
I would suggest to reserve the key as a prefix when its value is a dict.

Thanks,  
Bin

Reply:
The purpose is to keep only one if there are duplicate keys.

Reply:
I am not sure if users want this. Some of them may want to keep all metrics. But we can try your way first. If there is no bug report, maybe we can keep it.

Thanks  
Bin